lib/streamlit/watcher/event_based_path_watcher.py

```diff
@@ -360,6 +360,14 @@ def handle_path_change_event(self, event: events.FileSystemEvent) -> None:
         abs_changed_path = os.path.abspath(changed_path)
 
         changed_path_info = self._watched_paths.get(abs_changed_path, None)
+        if changed_path_info is None:
```

Comment:
This needed to be done since the event_based_path_watcher only checks for abs_changed_paths, this approach checks if the path that has changed is a common ancestor of the parent path. making it simple to check for for changes to a directory no matter the file under the directory